components/nimbus/src/behavior.rs

```diff
@@ -273,338 +287,44 @@ impl EventStore {
-        let store = EventStore::try_from(&db)?;
-        dbg!("From persisted data: {:?}", &store);
-
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Months)
-                .unwrap()
-                .data
-                .buckets
-                .len(),
-            1
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Months)
-                .unwrap()
-                .data
-                .buckets[0],
-            1
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets
-                .len(),
-            3
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets[0],
-            1
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets[1],
-            0
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets[2],
-            0
-        ));
-
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-2".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets
-                .len(),
-            1
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-2".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets[0],
-            0
-        ));
-
-        Ok(())
+    pub fn query(
+        &self,
+        event_id: String,
+        interval: Interval,
+        num_buckets: usize,
+        starting_bucket: usize,
+        query_type: EventQueryType,
+    ) -> Result<u64> {
```

Comment:
So Travis and I chatted about this, I had initially written the function to return a float. We landed on using integers and letting the default truncation behavior do its thing -- the intent being that it might be easier to document.

Reply:
FWIW: JEXL casts numbers to `f64`.

>  it might be easier to document.

I think there is a danger of being easier for the engineer to document, but harder for the end user to reason about.

e.g. the JEXL expression:

```jexl
'event'|my_query_function() > 2.5
```

would be incorrectly`false` if:

```
2.5 <= 'event'|my_query_function() < 3.0
```

Reply:
> So I've already done the jexl work for the next PR and I did indeed notice that. I ended up casting the f64->u64s but I do agree it might make more sense to have the query result be a float. @travis79 do you have any thoughts?

No objections to changing it from me.

Reply:
Okay I'll get that changed in the next PR if that's okay -- the next one is where it'll really come into play with the jexl work.